src/pages/JobLogs.tsx

```diff
@@ -0,0 +1,139 @@
+import { useEffect } from "react";
+import { useQuery, useLazyQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import Card from "@leafygreen-ui/card";
+import { H2, Subtitle, Body } from "@leafygreen-ui/typography";
+import { Skeleton } from "antd";
+import { useParams } from "react-router-dom";
+import { useJobLogsAnalytics } from "analytics/joblogs/useJobLogsAnalytics";
+import { StyledRouterLink, StyledLink } from "components/styles";
+import { getLobsterTestLogUrl } from "constants/externalResources";
+import { getTaskRoute } from "constants/routes";
+import { useToastContext } from "context/toast";
+import {
+  GetDisplayTaskQuery,
+  GetDisplayTaskQueryVariables,
+  GetTestsQuery,
+  GetTestsQueryVariables,
+} from "gql/generated/types";
+import { GET_DISPLAY_TASK, GET_TESTS } from "gql/queries";
+import { usePrevious } from "hooks";
+import { PageDoesNotExist } from "pages/404";
+
+export const JobLogs = () => {
+  const dispatchToast = useToastContext();
+  const { sendEvent } = useJobLogsAnalytics();
+  const { taskId, execution: execStr, groupId } = useParams<{
+    taskId: string;
+    execution: string;
+    groupId: string;
+  }>();
+  const execution = parseInt(execStr, 10);
+  const {
+    data: displayTaskResult,
+    loading: isLoadingDisplayTask,
+    error: errorDisplayTask,
+  } = useQuery<GetDisplayTaskQuery, GetDisplayTaskQueryVariables>(
+    GET_DISPLAY_TASK,
+    {
+      variables: { taskId, execution },
+      onError: (err) =>
+        dispatchToast.error(
+          `There was an error loading the task: ${err.message}`
+        ),
+    }
+  );
+  const [
+    getTests,
+    { data: testsResult, loading: isLoadingTests, error: errorTests },
+  ] = useLazyQuery<GetTestsQuery, GetTestsQueryVariables>(GET_TESTS, {
+    onError: (err) =>
+      dispatchToast.error(
+        `There was an error loading test result data: ${err.message}`
+      ),
+  });
+
+  const task = displayTaskResult?.task?.displayTask ?? displayTaskResult?.task;
+  const prevTask = usePrevious(task);
+  useEffect(() => {
+    if (!prevTask && task) {
+      getTests({
+        variables: {
+          taskId: task.id,
+          execution: task.execution,
+          groupId,
+        },
+      });
+    }
+  }, [task, prevTask, groupId, getTests]);
+
+  if (errorTests || errorDisplayTask) {
+    return <PageDoesNotExist />;
+  }
+
+  return isLoadingDisplayTask || isLoadingTests ? (
+    <Skeleton paragraph={{ rows: 8 }} />
+  ) : (
+    <Row>
+      <Card>
+        <Column>
+          <>
+            <H2>
+              <StyledRouterLink to={getTaskRoute(task?.id)} data-cy="task-link">
+                {task?.id}
+              </StyledRouterLink>
```

Comment:
Can/should we use the task name instead of the id?

Reply:
this should remain as taskId since that's what users are used to in the Logkeeper Job Logs page ([example](https://logkeeper.mongodb.org/build/1c451f3fc6fbc956273be52eaa33e7ec))